dig.go

```diff
@@ -188,8 +231,8 @@ func (c *Container) Invoke(function interface{}, opts ...InvokeOption) error {
 	return nil
 }
 
-func (c *Container) provide(ctor interface{}) error {
-	n, err := newNode(ctor)
+func (c *Container) provide(ctor interface{}, opts provideOptions) error {
```

Comment:
Are you sure that `opts ...ProvideOption` is not flattened too soon if it's immediately converted to a new struct and propagated down through `opts provideOptions`? Seems like you had to pipe it through quite a few structs in the new form and change tests.

At the very least you wouldn't have to change the test signatures if it was kept as varargs for a while.

Reply:
I wanted to avoid the user-facing `ProvideOption` being passed to the underlying layers. This is a good place IMO to convert to an "internal" representation of the options which is easy to consume from the other parts of the library.

The other structs are mostly in anticipation of future options that don't necessarily map to result options. Plumbing resultOptions in from provide, or plumbing provideOptions down to result, both smell to me.